### PR TITLE
HSD8-418 Allow user to hide block title

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Allow users to hide block titles

# Needed By (Date)
- 10/24

# Urgency
- high

# Steps to Test

1. checkout this branch
1. composer update
1. sync archaeology
1. edit one of the blocks in the footer
1. check the box "Hide Block Description"
1. save and validate block title is gone. 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by pookmish** [comment]
merge

File: docroot/modules/humsci/hs_blocks/hs_blocks.module [hunk 1]
```diff
@@ -34,3 +34,17 @@ function hs_blocks_theme($existing, $type, $theme, $path) {
     ],
   ];
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function hs_blocks_preprocess_block__block_content(&$variables) {
```
Comment (sherakama) [inline docroot/modules/humsci/hs_blocks/hs_blocks.module:41]: Double underscore?

File: docroot/modules/humsci/hs_blocks/hs_blocks.module [hunk 2]
```diff
@@ -34,3 +34,17 @@ function hs_blocks_theme($existing, $type, $theme, $path) {
...
+  /** @var \Drupal\block_content\Entity\BlockContent $block_entity */
+  $block_entity = $variables['content']['#block_content'];
```
Comment (sherakama) [inline docroot/modules/humsci/hs_blocks/hs_blocks.module:43]: Is this always going to be available? Should you check that the array key's are set or do you want the errors if they are not?

File: docroot/modules/humsci/hs_blocks/hs_blocks.module (hunk 1 above)
Comment (pookmish) [reply docroot/modules/humsci/hs_blocks/hs_blocks.module:41]: you can target specific hook ids to be more specific in the preprocess. `preprocess_block__BLOCK_ID`

File: docroot/modules/humsci/hs_blocks/hs_blocks.module (hunk 2 above)
Comment (pookmish) [reply docroot/modules/humsci/hs_blocks/hs_blocks.module:43]: Since its preprocess on a block content entity it will always be there

**Review by sherakama — APPROVED** [review]

**Pull request merged**